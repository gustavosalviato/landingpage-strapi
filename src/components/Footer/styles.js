import styled, { css } from "styled-components"
import { Text } from '../TextComponent/styles';

export const Container = styled.div`
    ${({ theme }) => css`
            text-align: center;
            border-top: 0.1rem solid #DDD;

            a{
                text-decoration: none;
            }            

            ${Text}{
                font-size: ${theme.font.sizes.small};
            }      

           
    `}
`







