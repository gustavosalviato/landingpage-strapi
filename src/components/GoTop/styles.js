import styled, { css } from "styled-components";

export const Container = styled.a`
    ${({ theme, background }) => css`
        position: fixed;
        display: flex;
        align-items: center;
        justify-content: center;
        background-color: ${theme.colors.darkBg};
        color: ${theme.colors.lightBg};
        width: 4rem;
        height: 4rem;
        bottom: 4rem;
        right: 4rem;
        z-index: 6;
        `}
`