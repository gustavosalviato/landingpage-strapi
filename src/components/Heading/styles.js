import styled, { css } from 'styled-components'

export const titleSize = {
    small: (theme) => css`
        font-size: ${theme.font.sizes.small}
    `,

    medium: (theme) => css`
        font-size: ${theme.font.sizes.medium}
    `,

    large: (theme) => css`
        font-size: ${theme.font.sizes.large}
    `,
};

const titleCase = (uppercase) => css`
    text-transform: ${uppercase ? 'uppercase' : 'none'};
`;



export const Title = styled.h1`
    ${({ theme, size, uppercase }) => css`
        color: ${theme.colors.lightBg};
        ${titleSize[size](theme)};
        ${titleCase(uppercase)}
    `}; 

    

`;