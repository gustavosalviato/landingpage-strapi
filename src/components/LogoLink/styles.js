import styled, { css } from "styled-components";


export const Container = styled.a`
    ${({ theme }) => css`
        text-decoration: none;
        color: inherit;

        >img{
            height: 6rem;
        }
    `}
`