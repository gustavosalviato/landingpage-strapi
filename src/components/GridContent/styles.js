import styled, { css } from "styled-components";
import { Title } from '../Heading/styles';
import { Text } from '../TextComponent/styles';
export const Container = styled.div`
    ${({ theme, background }) => css`
        max-width: 58rem;
        text-align: center;
        margin: 0 auto;

        ${Title}{
            color: ${background ? theme.colors.darkBg : theme.colors.lightBg};
        }
    `}
`

export const Html = styled.div`
    ${({ theme, background }) => css`
        margin: ${theme.spacings.xlarge} 0;

        ${Text}{
        color: ${background ? theme.colors.darkBg : theme.colors.lightBg};        
    }
    `}

   
`
