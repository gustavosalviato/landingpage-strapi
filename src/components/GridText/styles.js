import styled, { css } from 'styled-components'
import { Title } from '../Heading/styles'
import { Text } from '../TextComponent/styles'
export const Container = styled.div`
    ${({ theme }) => css`
        ${Text}{
            margin-bottom: ${theme.spacings.small};;
        }

    `}
`
export const Grid = styled.div`
    ${({ theme }) => css`
        counter-reset: grid-counter;
        display: grid;
        grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
        gap: ${theme.spacings.medium};
        overflow: hidden;
        width: 100%;

        @media ${theme.media.lteMedium}{
            grid-template-columns: 1fr;
            
        }
    `}
`

export const GridElement = styled.div`
    ${({ theme }) => css`
        ${Title}{
            position: relative;
            left: 5rem
        }
    

        ${Title}::before{
            counter-increment: grid-counter;
            content: counter(grid-counter);
            position: absolute;
            left: -5rem;
            font-size: 6rem;
            top: -2rem;
            transform: rotate(10deg);
        }
    
    `}
`



