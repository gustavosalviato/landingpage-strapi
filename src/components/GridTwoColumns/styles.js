import styled, { css } from "styled-components";
import { Title } from '../Heading/styles';

export const Container = styled.div`
    ${({ theme }) => css`
        display: grid;
        grid-template-columns: 1fr 2fr;
        align-items: center;
        gap: ${theme.spacings.large};

        @media ${theme.media.lteMedium}{
            grid-template-columns: 1fr;
            text-align: center;
        }

        ${Title}{
            margin-bottom: ${theme.spacings.large}
        }
    `}

`

export const TextContainer = styled.div`
    ${({ theme }) => css`
        @media ${theme.media.lteMedium}{
            margin-bottom: ${theme.spacings.large};
        }
        
    `}

`


export const ImageContaner = styled.div`
    ${({ theme }) => css`
        width: 600px;
        height: 600px;
    `}

`


export const Image = styled.img`
    ${({ theme }) => css`
        width: 100%;
        fill: red;
    `}

`