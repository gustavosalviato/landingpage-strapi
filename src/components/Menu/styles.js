import styled, { css } from "styled-components";
import { Container as SectionContainer } from '../SectionContainer/styles'
import { Title as Heading } from '../Heading/styles';

const menuVisible = (theme) => css`
    visibility: visible;
    opacity: 1;
`;

//Style for Mobile
export const Container = styled.div`
 ${({ theme, visible }) => css`
    position: fixed;
    z-index: 5;
    bottom: 0;
    top: 0;
    left: 0;
    right: 0;
    width: 100%;
    background: ${theme.colors.darkBg};
    transition: all 300ms ease-in-out;
 
    >${SectionContainer}{
        padding-top: 0;
        padding-bottom: 0;
    }


    & ${Heading}{
        padding-top: 0;
        padding-bottom: 0;
    }

    @media ${theme.media.lteMedium}{
    height: 100vh;
    visibility: hidden;
    opacity: 0;
    ${visible && menuVisible(theme)}

    > ${SectionContainer}{
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr;
        height: 100vh;
        align-items: center;
        overflow-y: auto;
        scroll-behavior: smooth;

    }


    & ${Heading}{
        display: flex;
        justify-content: center;
        padding-bottom: ${theme.spacings.large};
    }

 }`}
 
`;

export const MenuContainer = styled.div`
    ${({ theme }) => css`
        display: flex;
        justify-content: space-between;
        align-items: center;


        @media ${theme.media.lteMedium} {
        display: block;
        text-align: center;
        padding: ${theme.spacings.large} 0;
    }
    `}


`;

export const Button = styled.button`
    ${({ theme, visible }) => css`
        z-index: 6;
        position: fixed;
        top: 2rem;
        right: 2rem;
        width: 4rem;
        height: 4rem;
        background: ${theme.colors.darkBg};
        color: ${theme.colors.lightBg};
        border: none;
        display: none;
        cursor: pointer;
        pointer-events: ${visible ? 'none' : 'all'};

        @media ${theme.media.lteMedium}{
            display: flex;
            justify-content: center;
            align-items: center;
        }
    `}

    >svg{
        width: 2.5rem;
        height: 2.5rem;
    }


`;
