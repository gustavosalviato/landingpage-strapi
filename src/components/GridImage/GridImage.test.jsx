import { renderTheme } from "../../styles/render-theme";
import { screen } from '@testing-library/react';
import mock from './mock';
import { GridImage } from ".";
describe('<GridImage />', () => {
    it('should render grid image with background', () => {
        const { container } = renderTheme(<GridImage {...mock} background />);

        expect(container).toMatchSnapshot();

    })


    it('should render grid image without background', () => {
        const { container } = renderTheme(<GridImage {...mock} background={undefined} />);

        expect(container).toMatchSnapshot();

    })

  
});




