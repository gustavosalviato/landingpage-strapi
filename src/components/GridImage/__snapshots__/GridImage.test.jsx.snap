// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridImage /> should render grid image with background 1`] = `
<div>
  <div
    class="sc-eCYdqJ jdVqrb"
  >
    <div
      class="sc-jSMfEi yhBnn"
    >
      <div
        class="sc-bczRLJ dhcUoc"
      >
        <h1
          class="sc-gKXOVf hiRyZM"
        >
          My grid
        </h1>
        <p
          class="sc-iBkjds jwRHie"
        >
          Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
        </p>
        <div
          class="sc-gsnTZi cnoQKq"
        >
          <div
            class="sc-dkzDqf cVVNfV"
          >
            <img
              alt="https://source.unsplash.com/random/800x800?r=1"
              class="sc-hKMtZM jkrUDF"
              src="https://source.unsplash.com/random/800x800?r=1"
            />
          </div>
          <div
            class="sc-dkzDqf cVVNfV"
          >
            <img
              alt="https://source.unsplash.com/random/800x800?r=2"
              class="sc-hKMtZM jkrUDF"
              src="https://source.unsplash.com/random/800x800?r=2"
            />
          </div>
          <div
            class="sc-dkzDqf cVVNfV"
          >
            <img
              alt="https://source.unsplash.com/random/800x800?r=3"
              class="sc-hKMtZM jkrUDF"
              src="https://source.unsplash.com/random/800x800?r=3"
            />
          </div>
          <div
            class="sc-dkzDqf cVVNfV"
          >
            <img
              alt="https://source.unsplash.com/random/800x800?r=4"
              class="sc-hKMtZM jkrUDF"
              src="https://source.unsplash.com/random/800x800?r=4"
            />
          </div>
          <div
            class="sc-dkzDqf cVVNfV"
          >
            <img
              alt="https://source.unsplash.com/random/800x800?r=5"
              class="sc-hKMtZM jkrUDF"
              src="https://source.unsplash.com/random/800x800?r=5"
            />
          </div>
          <div
            class="sc-dkzDqf cVVNfV"
          >
            <img
              alt="https://source.unsplash.com/random/800x800?r=6"
              class="sc-hKMtZM jkrUDF"
              src="https://source.unsplash.com/random/800x800?r=6"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridImage /> should render grid image without background 1`] = `
<div>
  <div
    class="sc-eCYdqJ mMtnj"
  >
    <div
      class="sc-jSMfEi yhBnn"
    >
      <div
        class="sc-bczRLJ dhcUoc"
      >
        <h1
          class="sc-gKXOVf hiRyZM"
        >
          My grid
        </h1>
        <p
          class="sc-iBkjds jwRHie"
        >
          Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
        </p>
        <div
          class="sc-gsnTZi cnoQKq"
        >
          <div
            class="sc-dkzDqf cVVNfV"
          >
            <img
              alt="https://source.unsplash.com/random/800x800?r=1"
              class="sc-hKMtZM jkrUDF"
              src="https://source.unsplash.com/random/800x800?r=1"
            />
          </div>
          <div
            class="sc-dkzDqf cVVNfV"
          >
            <img
              alt="https://source.unsplash.com/random/800x800?r=2"
              class="sc-hKMtZM jkrUDF"
              src="https://source.unsplash.com/random/800x800?r=2"
            />
          </div>
          <div
            class="sc-dkzDqf cVVNfV"
          >
            <img
              alt="https://source.unsplash.com/random/800x800?r=3"
              class="sc-hKMtZM jkrUDF"
              src="https://source.unsplash.com/random/800x800?r=3"
            />
          </div>
          <div
            class="sc-dkzDqf cVVNfV"
          >
            <img
              alt="https://source.unsplash.com/random/800x800?r=4"
              class="sc-hKMtZM jkrUDF"
              src="https://source.unsplash.com/random/800x800?r=4"
            />
          </div>
          <div
            class="sc-dkzDqf cVVNfV"
          >
            <img
              alt="https://source.unsplash.com/random/800x800?r=5"
              class="sc-hKMtZM jkrUDF"
              src="https://source.unsplash.com/random/800x800?r=5"
            />
          </div>
          <div
            class="sc-dkzDqf cVVNfV"
          >
            <img
              alt="https://source.unsplash.com/random/800x800?r=6"
              class="sc-hKMtZM jkrUDF"
              src="https://source.unsplash.com/random/800x800?r=6"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
